src/circleci/rollcage/json.clj

```diff
@@ -0,0 +1,44 @@
+(ns circleci.rollcage.json
+  (:require
+    [clojure.string :as string]
+    [clojure.walk :refer (postwalk)]
+    [cheshire.core :as json]
+    [cheshire.generate :as gen]
+    [cheshire.factory :as factory])
+  (:import
+   [java.io StringWriter
+            Writer]
+   [com.fasterxml.jackson.core JsonFactory
+                               JsonGenerator
+                               JsonGenerationException]))
+
+(defn snake-case
+  [kw]
+  (string/replace (name kw) "-" "_"))
+
+(defn backstop-encoder
+  "Stringify an item if it's not possible to encode it normally. This should not
+   be called for most encoding jobs, only when it would otherwise throw an
+   exception."
+  [item]
+  (let [jg (.createGenerator
+            ^JsonFactory (or factory/*json-factory* factory/json-factory)
+            ^Writer (StringWriter.))]
+    (try
+     (gen/generate jg item factory/default-date-format nil snake-case)
+     item
+     (catch JsonGenerationException _
+       (str item)))))
```

Comment:
minor nit `(pr-str item)` might make thinks nicer (afair it will force lazy values, etc)